iface:: No need to validate absent interfaces

cathay4t opened the pull request:
When user are marking certain interface as absent, there is no need
to any more validation on this interface.

Integration test case added.

ffmancera commented on libnmstate/ifaces/base_iface.py:
```diff
@@ -222,14 +222,15 @@ def pre_edit_validation_and_cleanup(self):
-            ip_state = self.ip_state(family)
-            ip_state.remove_link_local_address()
-            self._info[family] = ip_state.to_dict()
+            if not self.is_absent:
+                for family in (Interface.IPV4, Interface.IPV6):
+                    self.ip_state(family).validate(
+                        IPState(family, self._origin_info.get(family, {}))
+                    )
+                self._validate_port_ip()
+                ip_state = self.ip_state(family)
+                ip_state.remove_link_local_address()
+                self._info[family] = ip_state.to_dict()
```
Indeed, I think this applies for any kind of validation, right? When marking an interface as absent we should not validate anything.

Did you check if there are other validations being executed when the interface is being marked as absent?

cathay4t replied:
I have removed all the validations on absent interface. Please check the new patch.

cathay4t committed:
iface: No need to validate absent interfaces

When user are marking certain interface as absent, there is no need
to any more validation on this interface.

Integration test case added.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera reviewed (approved):
LGTM, thanks!
